### PR TITLE
Adds *spin emote to humans and cyborgs, adds a 5 second cooldown to the Dog's tail chase verb

### PR DESCRIPTION
**Pull request by PidgeyThePirate**
## What Does This PR Do
Adds *spin and *spins to Human and Cyborg emote lists on a 5 second cooldown. Humans using *spin have a 5% chance of spinning too long and making themselves Dizzy and Confused for a short time.
Also adds a 5 second cooldown on the dog's tail chasing message to prevent text spam (the ability to actually chase one's tail is unimpeded).
Also rewrites the spin() function to spin either clockwise or counterclockwise.

## Why It's Good For The Game
Mob go spinny. Might be used as a dance move or in a similar range of cases as *flip.

## Images of changes
spinny

## Changelog
:cl:
add: Humanoids and Cyborgs can now use the *spin emote. Humanoids have a small chance of spinning too long and getting dizzy.
tweak: Dogs (particularly Ian) now have a 5 second cooldown on the chat message for chasing their tail.
/:cl:

### CODE REVIEW

File: code/modules/mob/living/carbon/human/emote.dm [hunk 1]
```diff
@@ -443,6 +445,17 @@
 								message = "<B>[src]</B> does a flip!"
 								SpinAnimation(5,1)
 
+		if("spin", "spins")
+			if(!restrained() && !lying)
+				if(prob(5))
+					INVOKE_ASYNC(src, .proc/spin, 30, 1)
```
Comment (farie82) [inline code/modules/mob/living/carbon/human/emote.dm:451]: It seems that `spin` itself has `waitfor = 0` meaning you don't need to call it async here. Same for below

**Comment by FlattestGuitar** [comment]
You only ever spin clockwise. This is a significant oversight.

File: code/modules/mob/mob.dm [hunk 2]
```diff
@@ -1343,24 +1343,52 @@ GLOBAL_LIST_INIT(slot_equipment_priority, list( \
...
-			if(SOUTH)
-				D = WEST
-			if(EAST)
-				D = SOUTH
-			if(WEST)
-				D = NORTH
-		setDir(D)
-		spintime -= speed
+	var/S
+	if(spin_direction == SPIN_DIRECTION_CLOCKWISE || spin_direction == SPIN_DIRECTION_COUNTERCLOCKWISE)
```
Comment (FlattestGuitar) [inline code/modules/mob/mob.dm:1356]:
```
var/end_time = worl.time + spintime
var/spin_dir = prob(50)
while(world.time < end_time)
  sleep(speed)
  if(spin_dir)
    dir = turn(dir, 90)
  else 
    dir = turn(dir, -90)
```

?

Comment (PidgeyThePirate) [reply code/modules/mob/mob.dm:1356]: I'll try it and see how it goes.

**Comment by Necaladun** [comment]
It's very cute, but I think we have enough silly stuff like this. All it will add to the game is more people acting unrealistically, which we have enough of.

**Comment by Dafrek** [comment]
Now we just need to be able to combo emotes so you can flip and spin at the same time.

**Review by S34NW — APPROVED** [review]

**Review by Kyep — APPROVED** [review]

**Review by variableundefined — APPROVED** [review]

**Pull request merged**